packages/apps/transfer/src/pages/transfer/cross-chain-transfer-finisher/index.tsx

```diff
@@ -1,128 +1,308 @@
 
-  const chainNetwork: {
-    Mainnet: { server: string; network: string };
-    Testnet: { server: string; network: string };
-  } = {
-    Mainnet: {
-      server: kadenaConstants.MAINNET.API,
-      network: kadenaConstants.MAINNET.NETWORKS.MAINNET01,
-    },
-    Testnet: {
-      server: kadenaConstants.TESTNET.API,
-      network: kadenaConstants.TESTNET.NETWORKS.TESTNET04,
-    },
-  };
-
   const [requestKey, setRequestKey] = useState<string>('');
   const [kadenaXChainGas, setKadenaXChainGas] =
     useState<string>('kadena-xchain-gas');
   const [gasPrice, setGasPrice] = useState<number>(0.00000001);
-  const [gasLimit, setGasLimit] = useState<number>(kadenaConstants.GAS_LIMIT);
   const [advancedOptions, setAdvancedOptions] = useState<boolean>(false);
+  const [showMore, setShowMore] = useState<boolean>(false);
+  const [pollResults, setPollResults] = useState<ITransferDataResult>({});
+  const [finalResults, setFinalResults] = useState<ITransferResult>({});
+  const [txError, setTxError] = useState('');
+
+  useEffect(() => {
+    setRequestKey('');
+    setPollResults({});
+    setFinalResults({});
+    setTxError('');
+  }, [network]);
 
-  const handleSubmit: FormEventHandler = (e) => {
+  const checkRequestKey = async (
+    e: React.KeyboardEvent<HTMLInputElement>,
+  ): Promise<void> => {
     e.preventDefault();
 
-    console.log('submitted');
+    if (!requestKey) {
+      return;
+    }
+
+    setFinalResults({});
+    setTxError('');
+
+    const pollResult: ITransferDataResult | undefined = await getTransferData({
+      requestKey,
+      server: chainNetwork[network].server,
+      networkId: chainNetwork[network].network as ChainwebNetworkId,
+      t,
+    });
+
+    if (pollResult === undefined) {
+      return;
+    }
+
+    setPollResults(pollResult);
+    if (pollResults.tx === undefined) {
+      return;
+    }
   };
 
+  const handleSubmit = async (
+    e: React.FormEvent<HTMLFormElement>,
+  ): Promise<void> => {
+    e.preventDefault();
+
+    if (!pollResults.tx) {
+      return;
+    }
+
+    const host = generateApiHost(
+      chainNetwork[network].server,
+      chainNetwork[network].network,
+      pollResults.tx.receiver.chain,
+    );
+
+    const contCommand = await finishXChainTransfer(
+      requestKey,
+      pollResults.tx.step,
+      pollResults.tx.pactId,
+      pollResults.tx.rollback,
+      chainNetwork[network].server,
+      chainNetwork[network].network as ChainwebNetworkId,
+      pollResults.tx.receiver.chain as ChainId,
+      kadenaXChainGas,
+    );
+
+    if (!(contCommand instanceof ContCommand) && contCommand.error) {
+      setTxError(contCommand.error);
+    }
+
+    if (contCommand instanceof ContCommand) {
+      try {
+        const pollResult = await contCommand.pollUntil(host, {
+          onPoll: async (transaction, pollRequest): Promise<void> => {
+            console.log(
+              `Polling ${requestKey}.\nStatus: ${transaction.status}`,
+            );
```

Comment:
I see various `console.log`s. I think it's nice to have a debuggable flow in the browser, whenever there's issues popping up. This could be done using the [`debug`](https://www.npmjs.com/package/debug) library. We could add relevant log statements across the xchain flow.  What do you think?